ArchiveHunterDownloadManager/AppDelegate.m

```diff
@@ -170,14 +170,27 @@ - (void)getUrl:(NSAppleEventDescriptor *)event withReplyEvent:(NSAppleEventDescr
                         totalSize = [fileSize longLongValue] + totalSize;
                     }
                     
-                    //NSLog(@"Total File Size: %lld", totalSize);
-                    [bulk setValue:[NSNumber numberWithLongLong:totalSize] forKey:@"totalSize"];
+                    NSDictionary *entryDict = [data objectForKey:@"entries"];
+                    NSUInteger keyCount = [entryDict count];
                     
-                    [[self managedObjectContext] save:&err];
-                    if(err){
-                        NSLog(@"could not save data store: %@", err);
+                    if(keyCount==0) {
+                        NSLog(@"There are no items to download!");
+                        NSAlert *alert = [[NSAlert alloc] init];
+                        [alert setMessageText:@"Download Error"];
+                        [alert setInformativeText:[NSString stringWithFormat:@"There are no items to download!"]];
+                        [alert addButtonWithTitle:@"Okay"];
+                        [alert runModal];
+                        [[self managedObjectContext] deleteObject:bulk];
+                    } else {
+                        //NSLog(@"Total File Size: %lld", totalSize);
+                        [bulk setValue:[NSNumber numberWithLongLong:totalSize] forKey:@"totalSize"];
+                    
+                        [[self managedObjectContext] save:&err];
```

Comment:
this line (and the error-handling afterwards) can be popped outside the if {} block, so the MOC gets saved whether we add update new bulk or delete it

Reply:
Done.